openstack_dashboard/api/glance.py

```diff
@@ -153,23 +157,25 @@ def __ne__(self, other_image):
         return not self.__eq__(other_image)
 
     def get_catalog_id(self):
-        if Image.PUBLISHED_APPLIANCES.get(self.id):
-            return Image.PUBLISHED_APPLIANCES.get(self.id).get('id', -1)
+        if cache.get('published_appliances', {}).get(self.id):
```

Comment:
I'm not sure what the `cache.get()` interface is like, but it may be actually trying to fetch from a remote cache (I hope not, but maybe.) In this case, it is definitely best to pull out the var early.

```python
appliances = cache.get('published_appliances', {})
return appliances.get(self.id, {}).get('id', -1)
```

Reply:
Hopefully `cache.get()` is set up to use memcached which lives in a container on the same host, so should be pretty quick. I like the simplified code you suggested, though I don't think there's an advantage in assigning cache to a local var here since the ref to the return value of `cache.get('published_appliances', {})` (cache value or empty dict) is maintained and used by the second get `.get(self.id, {}).get('id', -1)` so we could further simplify with, `return cache.get('published_appliances', {}).get(self.id, {}).get('id', -1)`